### ISSUE TITLE
twisted.internet.task.react should allow unprovided argv

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6389]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)| @Julian reported|
|-|-|
|Trac ID|trac#6389|
|Type|enhancement|
|Created|2013-03-24 22:03:28Z|
|Branch|https://github.com/twisted/twisted/tree/react-no-argv-6389|

Attachments:

* [react-argv.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/3c2/3c24f5d470b860351bc5e670aef38446c0a125dc/21da4eafc0530d274fc79d90c05133857806d583.patch) (2944 bytes) - added by Julian on 2013-03-24 22:03:36Z - 
* [react-argv.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/3c2/3c24f5d470b860351bc5e670aef38446c0a125dc/1cc12a7d634e8896b53ae61827f2846caf02cbf8.patch) (3442 bytes) - added by Julian on 2013-04-03 01:22:04Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__6389 6389
type__enhancement enhancement
reporter__Julian Julian
priority__normal normal
milestone__ 
branch__branches_react_no_argv_6389 branches/react-no-argv-6389
branch_author__jonathanj jonathanj
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1364162608000000 1364162608000000
changetime__1365416161000000 1365416161000000
version__None None
owner__jonathanj jonathanj

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)<a name="note_1"></a>|@Julian commented|
|-|-|

It seems unnecessary to need to do `react(main, [])`. Many APIs (including some in Twisted) that are delegating to other callables will use a default arg of `()` in place of `argv` if there aren't any args to pass along in a degenerate case, so it seems reasonable to expect to not need to pass it here.

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_2"></a>|@jonathanj set owner to @Julian|
|-|-|

  1. Please update the docstring for `argv`, possibly to mention that if a value is not provided it defaults to an empty tuple.

  2. Please add a news file.

Something that bothers me: There are no new tests for this and the old tests that used to pass `[]` have been changed. Maybe this is okay? But it's probably bothering me for a reason, I just don't know what it is.

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)<a name="note_3"></a>|@Julian removed owner|
|-|-|

Hey, thanks for the review.

Fixed the docstring. Had a topfile but I always forget `git add -N` so it didn't come through :/. Sorry.

It should bother you that now these tests don't cover the case where there's an arg provided, but I verified when making the change that there's another test that specifically test an arg being passed. Other than that, I think it's appropriate here.

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_4"></a>|@jonathanj commented|
|-|-|

(In [37965]) Branching to 'react-no-argv-6389'

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_5"></a>|@jonathanj commented|
|-|-|

(In [37966]) Apply react-argv2.patch, refs [#6389](https://github.com/twisted/twisted/issues/6389).

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_6"></a>|@jonathanj set owner to @Julian|
|-|-|

This looks good to merge.

[Build results](https://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/react-no-argv-6389)

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)|@Julian set owner to @jonathanj|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_8"></a>|@jonathanj set status to `closed`|
|-|-|

(In [37991]) Merge react-no-argv-6389

Author: Julian
Reviewer: jonathanj
Fixes: [#6389](https://github.com/twisted/twisted/issues/6389)

Allow omitting the "argv" argument, to mean no additional arguments, when calling twisted.internet.task.react.
